src/main/kotlin/uk/gov/justice/digital/hmpps/prisonregister/resource/PrisonEmailResource.kt

```diff
@@ -93,11 +96,50 @@ class PrisonEmailResource(private val prisonService: PrisonService) {
     @Size(max = 12, min = 2)
     prisonId: String,
   ): ResponseEntity<String> =
-    prisonService
-      .getOmuEmailAddress(prisonId)
+    prisonService.getEmailAddress(prisonId, OFFENDER_MANAGEMENT_UNIT)
       ?.let { ResponseEntity.ok(it) }
       ?: ResponseEntity.notFound().build()
 
+  @GetMapping(
+    "/$SECURE_PRISON_BY_ID/type/{contactPurposeType}/$EMAIL_ADDRESS",
+    produces = [MediaType.TEXT_PLAIN_VALUE],
+  )
+  @Operation(summary = "Get a prison's email address")
+  @ApiResponses(
+    value = [
+      ApiResponse(
+        responseCode = "200",
+        description = "Returns the email address",
+        content = [Content(mediaType = MediaType.TEXT_PLAIN_VALUE)],
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Client error - invalid prisonId or similar",
+      ),
+      ApiResponse(
+        responseCode = "404",
+        description = "The prison does not have a email address",
+      ),
+    ],
+  )
+  fun getEmailAddress(
+    @Schema(description = "Prison ID", example = "MDI", required = true)
+    @PathVariable
+    @Size(max = 12, min = 2)
+    prisonId: String,
+    @Schema(description = "ContactPurposeType", example = "social-visit or video-link-conferencing or offender-management-unit", required = true)
```

Comment:
remove example and move schema annotation to ContactPurposeType Enum?

Reply:
Yes not sure how to do this as enum is uppercase by convention and urls are not, will chat about this later